src/devices/src/virtio/mmio.rs

```diff
@@ -429,551 +332,552 @@ impl BusDevice for MmioDevice {
-            d.queue_select = q as u32;
-            byte_order::write_le_u32(&mut buf[..], 16);
-            d.write(0x38, &buf[..]);
-            byte_order::write_le_u32(&mut buf[..], 1);
-            d.write(0x44, &buf[..]);
-        }
-        assert!(d.are_queues_valid());
-        assert!(!d.device_activated);
-
-        // Device should be ready for activation now.
-        set_device_status(
-            d,
-            device_status::ACKNOWLEDGE
-                | device_status::DRIVER
-                | device_status::FEATURES_OK
-                | device_status::DRIVER_OK,
-        );
-        assert_eq!(
-            d.device_status,
-            device_status::ACKNOWLEDGE
-                | device_status::DRIVER
-                | device_status::FEATURES_OK
-                | device_status::DRIVER_OK
-        );
-        assert!(d.device_activated);
-    }
-
-    #[test]
-    fn test_bus_device_reset() {
-        let m = GuestMemoryMmap::from_ranges(&[(GuestAddress(0), 0x1000)]).unwrap();
-        let mut d = MmioDevice::new(m, Box::new(DummyDevice::new())).unwrap();
-        let mut buf = vec![0; 4];
-
-        assert!(!d.are_queues_valid());
-        assert!(!d.device_activated);
-        assert_eq!(d.device_status, 0);
-        activate_device(&mut d);
-
-        // Marking device as FAILED should not affect device_activated state
-        byte_order::write_le_u32(&mut buf[..], 0x8f);
-        d.write(0x70, &buf[..]);
-        assert_eq!(d.device_status, 0x8f);
-        assert!(d.device_activated);
-
-        // Nothing happens when backend driver doesn't support reset
-        byte_order::write_le_u32(&mut buf[..], 0x0);
-        d.write(0x70, &buf[..]);
-        assert_eq!(d.device_status, 0x8f);
-        assert!(d.device_activated);
-    }
-
-    #[test]
-    fn test_get_avail_features() {
-        let dummy_dev = DummyDevice::new();
-        assert_eq!(dummy_dev.avail_features(), dummy_dev.avail_features);
-    }
-
-    #[test]
-    fn test_get_acked_features() {
-        let dummy_dev = DummyDevice::new();
-        assert_eq!(dummy_dev.acked_features(), dummy_dev.acked_features);
-    }
-
-    #[test]
-    fn test_set_acked_features() {
-        let mut dummy_dev = DummyDevice::new();
-
-        assert_eq!(dummy_dev.acked_features(), 0);
-        dummy_dev.set_acked_features(16);
-        assert_eq!(dummy_dev.acked_features(), dummy_dev.acked_features);
-    }
-
-    #[test]
-    fn test_ack_features_by_page() {
-        let mut dummy_dev = DummyDevice::new();
-        dummy_dev.set_acked_features(16);
-        dummy_dev.set_avail_features(8);
-        dummy_dev.ack_features_by_page(0, 8);
-        assert_eq!(dummy_dev.acked_features(), 24);
-    }
-
-    #[test]
-    fn test_set_avail_features() {
-        let mut dummy_dev = DummyDevice::new();
-        assert_eq!(dummy_dev.avail_features(), 0);
-        dummy_dev.set_avail_features(4);
-        assert_eq!(dummy_dev.avail_features(), 4);
-    }
-}
+// #[cfg(test)]
```

Comment:
I'd rather remove the unit tests altogether, this is a lot of commented code.

Reply:
There's a pending PR #1609 that will make them work.

I propose we keep it like so when re-enabling tests, the diff will also show the changes made to the tests against the originals, should make it easier to review and/or check no tests are missed.